### ISSUE TITLE
Error while processing asynchronous task

### ISSUE THREAD

**Issue by ashishyadav13237** [Eernie/bitbucket-webhooks-plugin#85]
I am seeing this in the atlassian-bitbucket.log file. Could you please let me know what is causing this?

```
2017-12-08 11:23:11,283 ERROR [threadpool:thread-2] c.a.s.i.c.DefaultTransferableStateManager Error while processing asynchronous task
java.lang.NullPointerException: null
at nl.topicus.bitbucket.api.PushEventService.lambda$ignoredByBranch$0(PushEventService.java:54)
at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
at nl.topicus.bitbucket.api.PushEventService.ignoredByBranch(PushEventService.java:55)
at nl.topicus.bitbucket.api.PushEventService.isIgnoredEvent(PushEventService.java:30)
at nl.topicus.bitbucket.api.PushEventService.isValidEvent(PushEventService.java:26)
at nl.topicus.bitbucket.api.PullRequestListener.sendEvents(PullRequestListener.java:301)
at nl.topicus.bitbucket.api.PullRequestListener.lambda$onRepositoryRefsChanged$0(PullRequestListener.java:190)
at com.atlassian.stash.internal.concurrent.DefaultTransferableStateManager$StateTransferringRunnable.run(DefaultTransferableStateManager.java:166)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.lang.Thread.run(Thread.java:748)
... 1 frame trimmed
```

```
2017-12-08 12:17:32,094 ERROR [threadpool:thread-4] c.a.s.i.c.DefaultTransferableStateManager Error while processing asynchronous task
java.lang.NullPointerException: null
at nl.topicus.bitbucket.api.PushEventService.lambda$ignoredByBranch$0(PushEventService.java:54)
at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
at nl.topicus.bitbucket.api.PushEventService.ignoredByBranch(PushEventService.java:55)
at nl.topicus.bitbucket.api.PushEventService.isIgnoredEvent(PushEventService.java:30)
at nl.topicus.bitbucket.api.PushEventService.isValidEvent(PushEventService.java:26)
at nl.topicus.bitbucket.api.PullRequestListener.sendEvents(PullRequestListener.java:301)
at nl.topicus.bitbucket.api.PullRequestListener.lambda$onRepositoryRefsChanged$0(PullRequestListener.java:190)
at com.atlassian.stash.internal.concurrent.DefaultTransferableStateManager$StateTransferringRunnable.run(DefaultTransferableStateManager.java:166)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.lang.Thread.run(Thread.java:748)
... 1 frame trimmed
```

I am not sure what is causing this or how this can be reproduced. This is the FIRST time I am seeing this error in the log.

**Comment by casz** [comment]
Which version are you using @ashishyadav13237 ?

**Comment by ashishyadav13237** [comment]
I am running v1.6.0 with Bitbucket 5.5.1

Why don't i see v1.6.0 here?
https://marketplace.atlassian.com/plugins/nl.topicus.bitbucket.bitbucket-webhooks/versions

**Comment by casz** [comment]
Because of this exact error 👍

**Comment by casz** [comment]
@ashishyadav13237 please update to 1.6.2, should be all fixed and pretty

**Comment by ashishyadav13237** [comment]
Thanks. Just for my info, what is the cause of this error?

**Comment by casz** [comment]
Please read the release notes also pasted below 👍 

>#82 Fixes JSON marshal error
#81 Fixes Committers To Ignore Null Pointer
#78 Events are now only ignored if they are ignorable and conditions are true
#71 Added support for ignoring certain branches

**Comment by ashishyadav13237** [comment]
Thanks. I just upgraded to v.1.6.2

**Comment by casz** [comment]
Your issue was solved with #82 if any doubt
